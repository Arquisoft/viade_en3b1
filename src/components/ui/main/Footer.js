import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';
import GitHubIcon from '@material-ui/icons/GitHub';
import { Grid, Divider } from '@material-ui/core';
import LinkedInIcon from '@material-ui/icons/LinkedIn';
import EmailIcon from '@material-ui/icons/Email';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
  },
  main: {
    marginTop: theme.spacing(8),
    marginBottom: theme.spacing(2),
  },
  footer: {
    padding: theme.spacing(10, 10),
    marginTop: 'auto',
    backgroundColor:
      theme.palette.type === 'light' ? theme.palette.grey[200] : theme.palette.grey[800],
  },
}));

export default function StickyFooter() {
  const classes = useStyles();

  return (
    <div >
      <footer className={classes.footer}>
        <Container maxWidth="xs">
          <Grid container spacing={5}>
            <Grid item>
              <Typography variant="h6">ViaDe En3B1</Typography>
              <Typography variant="subtitle2" color="textSecondary">Made by <Link style={{ color: "#302c58" }} target="_blank" href={"https://www.linkedin.com/in/pablocanalsuarez/"}>Pablo Ca√±al</Link></Typography>
              <div>
                <Link href="https://www.linkedin.com/in/pablocanalsuarez/">
                  <LinkedInIcon fontSize="default" />
                </Link>
                <Link style={{ marginLeft: '0.5rem' }} target="_blank" href="mailto:uo264903@uniovi.es">
                  <EmailIcon fontSize="default" />
                </Link>
              </div>
            </Grid>
            <Divider orientation="vertical" flexItem />
            <Grid item style={{ marginTop: '1rem' }}>
              <Link target="_blank" href="https://github.com/Arquisoft/viade_en3b1/">
                <GitHubIcon fontSize="large" />
              </Link>
            </Grid>
          </Grid>
        </Container>
      </footer>
    </div>
  );
}