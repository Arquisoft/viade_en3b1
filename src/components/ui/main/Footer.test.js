import Footer from "./Footer";

test("Define Footer", () => {
    expect(Footer).toBeDefined();
});

test("Renders correctly", () => {
    const wrapper = render(<Footer />);
    expect(wrapper).toMatchSnapshot();
});