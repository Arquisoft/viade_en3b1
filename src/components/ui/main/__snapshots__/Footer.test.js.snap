// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <footer
    class="makeStyles-footer-3"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXs"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            ViaDe En3B1
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Made by 
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="https://www.linkedin.com/in/pablocanalsuarez/"
              style="color:#302c58"
              target="_blank"
            >
              Pablo Ca√±al
            </a>
          </h6>
          <div>
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="https://www.linkedin.com/in/pablocanalsuarez/"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                />
              </svg>
            </a>
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="mailto:uo264903@uniovi.es"
              style="margin-left:0.5rem"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </a>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
        />
        <div
          class="MuiGrid-root MuiGrid-item"
          style="margin-top:1rem"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://github.com/Arquisoft/viade_en3b1/"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
